### PR TITLE
release: fix Update method

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4355

The Update method was out of date. Now it's on par with Install. I'm
pretty sure it isn't used anywhere yet. Instead we use old Update method
from pkg/framework/resource.

### CODE REVIEW

File: pkg/release/release.go [hunk 1]
```diff
@@ -334,16 +334,22 @@ func (r *Release) InstallOperator(ctx context.Context, name string, chartInfo Ch
 	return nil
 }
 
-func (r *Release) Update(ctx context.Context, name, values, channel string, conditions ...func() error) error {
-	chartname := fmt.Sprintf("%s-chart", name)
+func (r *Release) Update(ctx context.Context, name string, chartInfo ChartInfo, values string, conditions ...func() error) error {
+	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
```
Comment (kopiczko) [inline pkg/release/release.go]: Even `releaseName` wasn't correct here.

**Review by xh3b4sd — APPROVED** [review]

**Comment by xh3b4sd** [comment]
Another thing. What can we do with the failing remote test right there?

**Comment by kopiczko** [comment]
> Another thing. What can we do with the failing remote test right there?

I don't know much about the cause yet. I took a note about that, but I still haven't looked into it.
Do you have ideas?

**Comment by xh3b4sd** [comment]
Disable it temporarily so we do not have to deal with the failing now. It seems to not be too useful anyway so this should be fine. We have it tracked and either remove it later or fix it.

**Review by MarcelMue — APPROVED** [review]

File: pkg/release/release.go [hunk 2]
```diff
@@ -257,16 +257,16 @@ func (r *Release) Install(ctx context.Context, name string, chartInfo ChartInfo,
...
 		return microerror.Mask(err)
 	}
 
-	err = r.helmClient.InstallFromTarball(tarball, r.namespace, helm.ReleaseName(releaseName), helm.ValueOverrides([]byte(values)), helm.InstallWait(true))
+	err = r.helmClient.InstallFromTarball(tarballPath, r.namespace, helm.ReleaseName(releaseName), helm.ValueOverrides([]byte(values)), helm.InstallWait(true))
 	if helmclient.IsReleaseAlreadyExists(err) {
 		return microerror.Maskf(releaseAlreadyExistsError, "failed to install release %#q", releaseName)
 	} else if helmclient.IsTarballNotFound(err) {
-		return microerror.Maskf(releaseAlreadyExistsError, "failed to install tarball %#q", tarball)
+		return microerror.Maskf(tarballNotFoundError, "failed to instll release %#q from tarball %#q", releaseName, tarballPath)
```
Comment (indyfree) [inline pkg/release/release.go]:
```suggestion
		return microerror.Maskf(tarballNotFoundError, "failed to install release %#q from tarball %#q", releaseName, tarballPath)
```

**Review by indyfree — APPROVED** [review]
Nice to get this sorted!

**Review by TheoBrigitte — APPROVED** [review]

**Comment by kopiczko** [comment]
I don't know how to disable this `e2eRemoteTest` test in the circle settings. @MarcelMue can you give me a hint? The workflow is fully green https://circleci.com/workflow-run/eab1f2a3-8a63-4be4-8539-ce1f6e45f5fc.

**Comment by MarcelMue** [comment]
I removed it, for the next time:
In github go to Settings -> Branches -> branch protection rules -> edit -> uncheck the test in the list -> save

**Pull request merged**